src/batch-branch/handle.ts

```diff
@@ -0,0 +1,39 @@
+import { OctokitParams } from '../../types';
+import { createBatchBranch } from './create';
+
+export async function handleBatchBranch(
+  { octokit, owner, repo }: OctokitParams,
+  batchPrAlreadyExists: boolean,
+  batchBranchName: string,
+  baseBranchName: string,
+) {
+  const allRefs = await octokit.paginate(octokit.rest.git.listMatchingRefs, {
+    owner,
+    repo,
+    ref: `heads/${batchBranchName}`,
+  });
+
+  // We have to filter for the exact branch name despite providing it to listMatchingRefs since it will return
+  // entries that begin with the branch name: https://octokit.github.io/rest.js/v18#git-list-matching-refs
+
+  const batchBranchAlreadyExists =
+    allRefs.filter(ref => ref.ref === `refs/heads/${batchBranchName}`).length === 1;
```

Comment:
Could replace `ref.ref` with `ref` by destructuring the ref param

Reply:
Nice one, thanks. Done :+1: